src/algorithm/hnswlib/hnswalg.cpp

```diff
@@ -705,7 +708,9 @@ HierarchicalNSW::resizeIndex(size_t new_max_elements) {
         throw std::runtime_error(
             "Cannot Resize, max element is less than the current number of elements");
 
-    visited_list_pool_.reset(new VisitedListPool(1, new_max_elements, allocator_));
+    auto new_visited_list_pool = allocator_->New<VisitedListPool>(new_max_elements, allocator_);
```

Comment:
i guess you want do this: 
```C++
auto old_visited_list_pool = visited_list_pool_;
visited_list_pool_ = allocator_->New<VisitedListPool>(new_max_elements, allocator_);
allocator_->Delete(old_visited_list_pool);
```

or direct delete old and create new one

```C++
allocator_->Delete(visited_list_pool_);
visited_list_pool_ = allocator_->New<VisitedListPool>(new_max_elements, allocator_);
```

Reply:
It cannot be modified this way. If we first delete visited_list_pool, subsequent memory allocation failures will lead to the index becoming unavailable.